apps/server/src/api-integration/integration.utils.ts

```diff
@@ -16,9 +17,35 @@ const whitelistedPayload = {
   skip: coerceBoolean,
 
   colour: coerceColour,
+
+  custom: coerceCustom,
 };
 
+function coerceCustom(value: unknown): EventCustomFields {
+  const customValue = value as { value: unknown; customProperty: string; type: 'string' };
+  switch (customValue.type) {
+    case 'string':
+      return {
+        [customValue.customProperty]: {
+          value: coerceString(customValue.value,
+          ),
+        },
+      };
+  }
+  throw new Error('Unable to parese custom value');
+}
```

Comment:
There seem to be many assumptions here. I think we need to think about this piece of code 

The assertion is done prematurely, we likely want a type guard instead

The switch statement offers no value here